python/tskit/cli.py

```diff
@@ -87,6 +87,11 @@ def run_upgrade(args):
             "will remove these duplicate positions. However, this will result in loss "
             "of data from the original file!"
         )
+    except OSError as e:
+        if "file signature not found" not in str(e):
+            raise e
+        else:
+            tskit.fix_legacy_kastore(args.source, args.destination)
```

Comment:
Hmmm, shouldn't this be done in ``tskit.load_legacy``? We want this function to load kastore files and fix them up as well, not just run the upgrade in the CLI.